runtime/src/message_processor.rs

```diff
@@ -341,10 +348,19 @@ impl<'a> InvokeContext for ThisInvokeContext<'a> {
         }
     }
     fn get_caller(&self) -> Result<&Pubkey, InstructionError> {
-        self.program_ids
+        self.invoke_stack
             .last()
+            .map(|frame| &frame.0)
             .ok_or(InstructionError::GenericError)
     }
+    fn remove_first_keyed_account(&mut self) {
+        let index = self.invoke_stack.len() - 1;
```

Comment:
`.last()` would be safer, if for whatever reason `.len()` was zero this would underflow

Reply:
Yes, will have to do that anyway to get past `clippy::integer-arithmetic`